AzureCalling/Controllers/AudioDeviceSelectionViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class AudioDeviceSelectionViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // MARK: Properties
+
+    private var audioDeviceOptions: [AudioDeviceOption] = [AudioDeviceOption]()
+    var deviceTable: UITableView!
+
+    // MARK: UIViewController events
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.modalTransitionStyle = .crossDissolve
+
+        view.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.8)
+        view.isOpaque = false
+
+        // tapping anywhere on the view is the same as tapping cancel
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissSelf))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+        view.isUserInteractionEnabled = true
+
+        createAudioDeviceOptions()
+        createDeviceTable()
+    }
+
+    @objc func dismissSelf(sender: NSObject) {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        openDeviceTable()
+    }
+
+    func createDeviceTable() {
+        deviceTable = UITableView()
+        deviceTable.isHidden = true
+        deviceTable.allowsSelection = true
+        deviceTable.isScrollEnabled = false
+        deviceTable.layer.cornerRadius = 8
+        deviceTable.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(deviceTable)
+        let audioDeviceCell = UINib(nibName: "BottomDrawerCellView",
+                                      bundle: nil)
+        deviceTable.register(audioDeviceCell, forCellReuseIdentifier: "BottomDrawerCellView")
+        deviceTable.dataSource = self
+        deviceTable.delegate = self
+        deviceTable.reloadData()
+
+        var deviceTableConstraints = [
+        deviceTable.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        deviceTable.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+        deviceTable.heightAnchor.constraint(equalToConstant: 100)
```

Comment:
we should separate a ticket to handle the dynamic height of the table view. And make the table view reusable for `participant view list`. 
We could call the audioSessionManager in the ViewController and pass the `cellViewData` into this table view. And then based on closure to pass the selectedData back to viewController. And this tableViewController has no idea what type of data it's displaying

Reply:
Done.  Modified it such that the table height is dynamic to its content size, and I also include proper margin to account for different iPhone curvatures.  Here's an iPhone SE:
![image](https://user-images.githubusercontent.com/6021109/116142259-8d3bb780-a68e-11eb-9f7a-c7b3d71c2490.png)
And iPhone 12 Pro:
![image](https://user-images.githubusercontent.com/6021109/116142290-988ee300-a68e-11eb-843a-5ab04a07a02d.png)